docs/DeveloperDoc.md

```diff
@@ -0,0 +1,15 @@
+# Developer Documentation
+
+## How to render a palette
+
+`Palette.ts` accepts a JSON in a structure demostrated in
+[`src/keyboards/bmw_palette.json`](../src/keyboards/bmw_palette.json). In the
+`cells` object, each cell has a `type` key and a `options` key. The `type`
+value indicates which component should be used to render this cell. The
+`options` contains information to be passed to the component.
+
+When a new `type` value is introduced, a developer need to:
```

Comment:
There is a mismatch between singular subject, "a developer", and plural verb, "need to".  I suggest using plural throughout:

"
When a new `type` value is introduced, developers need to:
"